src/attribute.cpp

```diff
@@ -14,6 +15,21 @@ String::String(uint8_t type, const uint8_t* data, size_t size)
 {
 }
 
+String::String(uint8_t type, const std::string& string)
+    : Attribute(type),
+      m_value(string)
+{
+}
+
+std::vector<uint8_t> String::toVector(const std::string& secret, const std::array<uint8_t, 16>& auth) const
```

Comment:
`secret` and `auth` are not used in this function. Keep the names, but comment them out:
```c++
std::vector<uint8_t> String::toVector(const std::string& /*secret*/, const std::array<uint8_t, 16>& /*auth*/) const
```

Same below.